regression/basic_flow.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Create a streaming dataset from toy data with various options for regression testing."""
+
+import os
+import shutil
+import tempfile
+from argparse import ArgumentParser, Namespace
+
+from torch.utils.data import DataLoader
+
+from streaming import StreamingDataset
+from streaming.base.distributed import barrier
+
+_TRAIN_EPOCHS = 2
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns:x
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--local', default=False, action='store_true')
+    args.add_argument(
```

Comment:
We can use something like this so that you don't have to copy/paste the argparse from multiple places to here.

```python
def get_kwargs(kwargs: List[str]) -> Dict[str, str]:
    if kwargs.startswith('--'):
        kwargs = kwargs[2:]
    kwargs = kwargs.replace('-','_')
    return kwargs

args, runtime_args = args.parse_known_args()
kwargs = {get_kwargs(k):v for k,v in zip(runtime_args[::2],runtime_args[1::2])}

# one can use it as
print(int(kwargs.get('batch_size', 32)))
print(kwargs.get('cache_limit', '30gb'))
```

This way you don't have to add `args.add_argument(` for `download_retry`, `download_timeout`, `validate_hash`, etc.